<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">

**Привет, Сергей!** 👋

Меня зовут Алексей Манелюк, я буду проверять твой проект. В Яндекс Практикуме принято общаться на "ты" 😉, но если некомфортно - скажи и перейдем на "Вы".

Моя цель - не указать на ошибки, а поделиться опытом и помочь в освоении анализа данных.

В тетрадке ты увидишь мои комментарии, пожалуйста, не удаляй, не перемещай и не изменяй их. У комментариев могут быть приписки **V2**, **V3**, **V...**, которые соответствуют каждой версии проверки. 
    
Я буду использовать цветовую разметку:
<p>
<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Так выделены критические замечания. Без их отработки я не смогу принять проект. </div>

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>  Комментарий ревьюера ⚠️</b>

Так выделены небольшие замечания или полезные советы. Постарайся учесть их.</div>
<p>
<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Так выделены правильные решения.</div>

Предлагаю работать в диалоге: если что-то дорабатываешь или отвечаешь на мой комментарий - пиши об этом. Мне будет удобней, если ты выделишь свои комментарии:
<p>
<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

Все это поможет мне быстрее принять твой проект.</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Несмотря на то, что первая часть проекта проверяется автоматизрованно, дам по ней пару советов - они пригодятся тебе в следующих проектах. 

**Их отработка сейчас необязательна**.

📌 После названия, обязательно нужно добавлять бриф, цели исследования, описание данных и план работы.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Алексей, добрый день! Спасибо, за проверку моего проекта! Замечания будут учтены для последующих работ! .</div>

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
#Проект для кредитного отдела банка. 
#Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. 
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()#После вывода информации, обнаружил, что есть пропуски в данных. При этом пропуски данных мы получили в равных количествах по столбцам days_employed и total_income

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

📌 Каждый раздел работы стоит завершать кратким выводом, о том что было сделано и какие результаты получены.

Как правильно оформить проект ты можешь прочитать в материалах курса:

`Полезные инструкции для учёбы - Оформление проекта - Рекомендации по выполнению проектов`
</div>

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()#Выводы подтвердились, имеет равное количество пропусков по двум столбцам

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():#заполнили пропуски total_income медианными значениями
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()#после вывода уникальных значений chilren получил результаты с отрицательным количеством детей и слишком большим

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]#удалил значения

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():#заполнили пропуски days_emplyed медианными значениями
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()#избавился от пропусков

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()#количество дубликатов оказалось 71, удалили

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):#создал категории клиентов по доходу, было разделение на 5 категорий
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
#data.info()# Ваш код будет здесь. Вы можете создавать новые ячейки.
#data.head(20)
question = data.pivot_table(index = ['children'], values = 'debt', aggfunc = {'count','sum'}) #debt-необходимо посчитать людей у которых есть задолжность
question['answer'] = round(question['sum'] / question['count'], 3) #Посчитаем какой процент людей имеет задолжность, для этого разделим количество людей с задолжностью на клиентов.
question = question.sort_values(by='answer', ascending=True) #сортируем и выводим по возрастанию
display(question)

count   sum  answer
children                     
5             9     0   0.000
0         14091  1063   0.075
3           330    27   0.082
1          4808   444   0.092
2          2052   194   0.095
4            41     4   0.098

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

📌 Для более аккуратного вывода таблиц в Jupyter Notebook, лучше использовать функцию `display()` вместо `print()`.

А если вывод таблицы в последней строке ячейки Jupyter Notebook, можно вообще эти функции не использовать.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Исправил.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Отлично!</div>

**Вывод:** Самый большой % задолжности у людей с 4-мя детьми, вероятно из-за высокой фин.нагрузки. почти одинаковые значения у людей с 1 и 2 детьми. Самый маленький % у людей с 5 детьми, вероятно из-за высокого уровня фин.дисциплины, хотя таких клиентов не так много.

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Ты правильно обратил внимание на то, что выборки несбалансированны. Поэтому выявлять зависимость стоит, сравнивая самые большие, сопоставимые по размеру категории заемщиков. Какую зависимость можно выявить, сравнив категории с 0, 1 и 2 детьми?</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Зависимость, между тремя этими группами следующая. Категория с 1 и 2 детьми имеет практически одинаковый процент по просрочке. Вероятно, это связано с тем, что на семье находится дополнительная фин.нагрузка из-за детей. Категория, не имеющая детей имеет меньшй % просрочки, так-как они меньшую финансовую нагрузку. Следовательно, люди имеющие детей относятся к группе риска не отдать кредит вовремя.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Если сравнить только самые многочисленные выборки с 0, 1 и 2 детьми, можно сказать, что чем больше у заемщика детей, тем выше риск допустить просрочку.</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
question = data.pivot_table(index = ['family_status'], values = 'debt', aggfunc = {'count','sum'}) #debt-необходимо посчитать людей у которых есть задолжность
question['answer'] = round(question['sum'] / question['count'], 3) #Посчитаем какой процент людей имеет задолжность, для этого разделим количество людей с задолжностью на клиентов.
question = question.sort_values(by='answer', ascending=True) #сортируем и выводим по возрастанию
display(question)

count  sum  answer
family_status                            
вдовец / вдова           951   63   0.066
в разводе               1189   84   0.071
женат / замужем        12261  927   0.076
гражданский брак        4134  385   0.093
Не женат / не замужем   2796  273   0.098

**Вывод:**  Самый высокий процент по просрочкам у людей с семейным статусом "не женат/ не замужем", вероятно из-за низкой фин.дисциплины, так и у людей в гражданском браке, вероятно это люди которые не ведут общего бюджета. Люди с статусом "женат/замужем" в середине списка, так-как имеют с одной стороны высокую фин.нагрузку на семью, с другой стороны имеют общий бюджет. Наименьший процент у людей с семейным статусом "вдовец/вдова" и "в разводе". Вероятно это люди которые находятся старше среднего возраста, наиболее обеспечены и имеют расходы в большей степени только на себя. 

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Верно, если обобщить, то можно сказать, что состоящие или состоявшие в браке заемщики реже допускают просрочки.</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
question = data.pivot_table(index = ['total_income_category'], values = 'debt', aggfunc = {'count','sum'}) #debt-необходимо посчитать людей у которых есть задолжность
question['answer'] = round(question['sum'] / question['count'], 3) #Посчитаем какой процент людей имеет задолжность, для этого разделим количество людей с задолжностью на клиентов.
question = question.sort_values(by='answer', ascending=True) #сортируем и выводим по возрастанию
display(question)


count   sum  answer
total_income_category                     
D                        349    21   0.060
B                       5014   354   0.071
A                         25     2   0.080
C                      15921  1353   0.085
E                         22     2   0.091

**Вывод:** Самый высокий уровень не возврата кредита у категории Е с ЗП до 30 000 рублей , так-как низкая ЗП плата всего ставит перед клиентом высокую фин. дисциплину, но любая жизненная ситуация с непредвидеными расходами поставит эту категорию клиентов в потенциальных должников. Что касается категории Е и А, это самый большие и самые высокие доходы по выборкам, количество таких пользователей крайне мало в % соотношении к другим клиентам. Поэтому по такому количеству клиентов выводы сделать крайне сложно, так-как статистической информации мало. Что касается категории С - это категория имеет большинство среди пользователей, эти пользователи как правило имеют не очень высокую фин. дисциплину, так-как имеют достаточно высокие фин.доходы среди населения РФ например. И вполне могут себе позволить задержать выплату, ввиду штраф для них будет не такой существенной потерей. Самый низкий % невозврата у категории D - эта категория вероятно обладает высоким уровнем дисциплины, так-как при достаточно невысокой зарплате скорее всего планирует свои расходы на месяц, и при этом такую категорию трудно вывести из стабильности незапланированными тратами.


<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

А какую можно выявить зависимость, если сравнить только две самые многочисленные выборки?</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Если сравнивать две самые многочисленные выборки, то группа B - имеет меньший % по задоженности, вероятно эти люди предприниматели, они имеют достаточно высокие доходы и вероятно следят за своей кредитной историей, тем самым имеют самую низий % по задолженности. Что касается самой многочисленной категории, категории С, то это большая часть населения, которая вероятно имеет несколько детей, возможно не один кредит и имеют высокую финансовую нагрузку, возможно некоторые клиенты в этой категории менее беспокоятся за свою кредитную историю.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Если сравнить самые большие категории B и C, можно сказать, что заемщики с более высоким доходом платят кредиты вовремя лучше.</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
question = data.pivot_table(index = ['purpose_category'], values = 'debt', aggfunc = {'count','sum'}) #debt-необходимо посчитать людей у которых есть задолжность
question['answer'] = round(question['sum'] / question['count'], 3) #Посчитаем какой процент людей имеет задолжность, для этого разделим количество людей с задолжностью на клиентов.
question = question.sort_values(by='answer', ascending=True) #сортируем и выводим по возрастанию
display(question)

count  sum  answer
purpose_category                            
операции с недвижимостью  10751  780   0.073
проведение свадьбы         2313  183   0.079
операции с автомобилем     4279  400   0.093
получение образования      3988  369   0.093

**Вывод:** самый высокий % по невозвратам на "операции с автомобилем" и "получение образования" - вероятно в этой категории присутствуют лица менее 30 лет, эта категория людей менее ответственна в финансовых вопросах. Что касаемо операцией с недвижимостью, здесь вероятно уже  люди с семьей, более ответственные к планированию бюджета, более состоятельные и имеют скорее всего общий бюджет семейный, все это благотворно сказывается на оплате кредита вовремя 

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Все правильно. Молодец, что формулируешь гипотезы о причинах, которые могли привести к полученным результатам. 👍</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

    *Ответ:* Пропуски были в 2 столбцах days_empleyed и total_income, самое важное, что количество пропусков в столбце было одинаковое количество. В проекте присутсвуют "категории безработных,студентов,в декрете" которые могли бы вместо этих ответов их просто напросто пропустить, из-за того что они на данном этапе равны = 0. Есть вероятность, что одна из категорий решила просто напросто не заполнять эти столбики, так-как их количество равно друг другу. 

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Ты правильно указал человеческий фактор (клиент не ввел данные), как наиболее вероятную причину образования пропусков в данных. Но какая причина еще может вызвать потерю данных?</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Причиной потери данных может быть не только сокрытие информации и неправильный/некорректный ввод. Но возможно ошибки в логике системы "сбора данных" или хранении. Следует рассмотреть оба варианта.  Если это первый случай, необходимо не пропускать пользователя дальше, пока он не заполнит корректно "строки ввода", если это второй случай, после анализа выгрузить данные с пропусками и принять решение по их исправлению.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Техническую ошибку также исключать нельзя. Они могут возникать во время копирования, считывания или записи данных, изменения формата файла.</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Арифметическое среднее значение очень грубо отражает данные, благодаря мединному способу мы можем более точно получить информацию и заполнить недостающие атрибуты, это есть в примере теории, когда директору по маркетингу решили платить 300 тыс.руб, а всем остальным по 100.

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Совершенно верно, медиана менее чувствительна к выбросам. Для примера можешь рассчитать эти метрики для столбца `days_employed`, где есть выбросы.</div>

### Шаг 4: общий вывод.

Исходя из полученных данных, мы можем предположить , что полноценные семьи с большим количеством детей и неплохим доходов - имеют более высокие риски вовремя не заплатить кредит, нежели категория людей которые не имеют детей, в любом статусе, но при высоких доходах. Самый высокий по риску вариант, если у вас более двух детей, вы не замужем/женаты, и находитесь в категории с ЗП (С). Такой человек имеет самый высокий % иметь задолженность. И наоборот человек без отношений, без детей, с доходом D имеет самые высокие шансы выплатить кредит вовремя. НО шанс, что такие выборки соберутся в единое целое крайне мал, поэтому карреляция прослеживается слабо.

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Общий вывод в целом верный, однако есть смысл немного его расширить. Нужно отразить в нем основные результаты по всем шагам работы. Начать с абзаца с описанием входных данных и что сделали на этапе предобработки. 

Далее описать все выявленные зависимости просрочки от различных факторов в общем виде. 

Закончить вывод можно рекомендациями банку по улучшению сбора данных.

**Зачастую, заказчик читает только общий вывод, поэтому он должен быть максимально информативным.**

Можно выделить самые важные моменты форматированием.

Вот ссылочка, которая поможет с форматированием текста в ячейке Markdown

https://medium.com/analytics-vidhya/the-ultimate-markdown-guide-for-jupyter-notebook-d5e5abf728fd</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Был проведен анализ данных для кредитного отдела банка, с целью улучшения скоринговой оценки на основе статистических данных, а именно "Влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок". Заказчиком были переданы статистические данные в формате CSV за период работы отдела. На этапе анализа данных на полноту исследования, были выявлены ряд логических ошибок, такие как:

1. В ряде данных отстутсовала информация по столбцам days_empleyed и total_income. В общей сложности, благодаря структурированию данных и их аналитике, было установлено 2174 пропуска в обоих столбцах. Для полноценного анализа данных, пропуски были заполнены средним значением по анализируемым данным, выведенным по медианному значению.
    
2. По анализируемым данным в столбцах children и education были неверные значения
    -children отрицательные и завышенные значения
    -education разные шрифты, дублирование данных
Исправлено.
    
На основе дополненных и структурированных данных, был выполнен анализ по вопросам:

    а) Есть ли зависимость между количеством детей и возвратом кредита в срок? (Наличие детей, является одним из факторов, что у заемщика будет просрочка.)
    б) Есть ли зависимость между семейным положением и возвратом кредита в срок? (состоящие или состоявшие в браке заемщики реже допускают просрочки.)
    в) Есть ли зависимость между уровнем дохода и возвратом кредита в срок? (Заёмщики были разбиты на категории по количеству получаемых средств, наиболее большая группа и имеющая большой % по просроченным кредитам, это люди с размером ЗП 50 000 - 200 000)
    г) Как разные цели кредита влияют на его возврат в срок?(самый высокий % по невозвратам на "операции с автомобилем" и "получение образования")

Рекомендации по поводу данных:
    
Причиной потери данных может быть не только сокрытие информации и неправильный/некорректный ввод. Но возможно ошибки в логике системы "сбора данных" или хранении. Следует рассмотреть оба варианта. Если это первый случай, необходимо не пропускать пользователя дальше, пока он не заполнит корректно "строки ввода", если это второй случай, после анализа выгрузить данные с пропусками и принять решение по их исправлению.
    
 Исходя из полученных данных, мы можем предположить , что полноценные семьи с большим количеством детей и неплохим доходов - имеют более высокие риски вовремя не заплатить кредит, нежели категория людей которые не имеют детей, в любом статусе, но при высоких доходах. Самый высокий по риску вариант, если у вас более двух детей, вы не замужем/женаты, и находитесь в категории с ЗП (С). Такой человек имеет самый высокий % иметь задолженность. И наоборот человек без отношений, без детей, с доходом D имеет самые высокие шансы выплатить кредит вовремя. НО шанс, что такие выборки соберутся в единое целое крайне мал, поэтому карреляция прослеживается слабо.
</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Теперь у тебя получился подробный общий вывод, отражающий все основные результаты исследования. 👌</div>

<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">
    
<b>Общий комментарий ревьюера</b>

С проектом все хорошо, ты здорово справился с заданием. 👌 
    
Критических замечаний у меня нет, постарайся только отработать желтые комментарии во второй части работы.
    
Буду ждать твой доработанный проект для принятия. ⌛️
    
</div>

<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">
    
<b>Общий комментарий ревьюера V2</b>

Спасибо, что доработал проект, получилась очень хорошая работа.
    
Успехов в дальнейшем обучении! 🏆
    
</div>